### PR TITLE
Zstd, README, logging

### PR DESCRIPTION
**Pull request by philandstuff**
This changes the existing `r8im uncompress` command to `r8im zstd`. It also updates the README to be clearer about what this tool does.

It also also adds logging for go-containerregistry operations.

### CODE REVIEW

**Comment by anotherjesse** [comment]
lgtm

File: README.md [hunk 1]
```diff
@@ -1,76 +1,75 @@
...
 
-- doesn't require or use docker / containers
-- works with image registry directly
-- attach weights (or other changes) to an existing image
-- modify existing images without downloading layers
-- works inside of docker
+This is a suite of assorted tools for inspecting and manipulating
+docker and OCI images for machine learning workloads.
 
-## story time: the nightmare
```
Comment (technillogue) [inline README.md:9]: I would love to keep the motivating story tbh

File: pkg/images/uncompress.go [hunk 2]
```diff
@@ -1,153 +0,0 @@
...
-	}
-
-	addendums := make([]mutate.Addendum, max(len(ocf.History), len(layers)))
-
-	startUncompressing := time.Now()
-
-	var historyIdx, addendumIdx int
-	for layerIdx := 0; layerIdx < len(layers); addendumIdx, layerIdx = addendumIdx+1, layerIdx+1 {
-		startLayer := time.Now()
-		newLayer, err := uncompressedLayer(layers[layerIdx])
```
Comment (technillogue) [inline pkg/images/uncompress.go:82]: should we also keep the uncompressed version, or did we decide that was strictly worse?

File: pkg/images/zstd.go [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+			return nil, fmt.Errorf("setting recompressed layer: %w", err)
+		}
+		// truncate time to 3 decimal places
+		truncTime := time.Duration(int64(time.Since(startLayer).Seconds()*1000)) * time.Millisecond
+		fmt.Fprintln(os.Stderr, "recompressing layer", layerIdx, "took", truncTime)
+		// uncompressedSize, err := newLayer.Size()
+		if err != nil {
+			return nil, fmt.Errorf("getting recompressed size: %w", err)
+		}
+		// compressionRatio := int(float64(compressedSize)/float64(uncompressedSize)*100) / 100.0
```
Comment (technillogue) [inline pkg/images/zstd.go:173]: do we want to keep a check for leaving layers with >1 CR  uncompressed, or check for weights layer?

**Pull request merged**